test/api/pulses_test.go

```diff
@@ -0,0 +1,220 @@
+			require.Equal(t, res.PulseNumber-10, res.PrevPulseNumber)
+			require.Equal(t, res.PulseNumber+10, res.NextPulseNumber)
+			require.Equal(t, recordsInPulse, int(res.JetDropAmount))
+			require.Equal(t, recordsInPulse, int(res.RecordAmount))
+			require.NotEmpty(t, res.Timestamp)
+		}
+	})
+	t.Run("valid limit and offset", func(t *testing.T) {
+		t.Log("C5209 Get pulses with limit and offset")
+		offset, limit := 3, 3
+		opts := client.PulsesOpts{Offset: optional.NewInt32(int32(offset)), Limit: optional.NewInt32(int32(limit))}
+		response, err := c.Pulses(t, &opts)
+		require.NoError(t, err)
+		require.Len(t, response.Result, limit)
+		require.Equal(t, pulses[len(pulses)-4], response.Result[0].PulseNumber)
+		require.Equal(t, pulses[len(pulses)-5], response.Result[1].PulseNumber)
+		require.Equal(t, pulses[len(pulses)-6], response.Result[2].PulseNumber)
+	})
+	t.Run("out of offset", func(t *testing.T) {
+		t.Log("C5178 Get pulses, offset > max pulses")
+		limit, offset := 10, pulsesNumber+1
+		opts := client.PulsesOpts{Limit: optional.NewInt32(int32(limit)), Offset: optional.NewInt32(int32(offset))}
+		response, err := c.Pulses(t, &opts)
+		require.NoError(t, err)
+		require.Equal(t, response.Total, int64(pulsesNumber))
+		require.Len(t, response.Result, 0)
+	})
+	t.Run("limit min", func(t *testing.T) {
+		t.Log("C5170 Success limit validation, limit = 1")
+		limit := 1
+		opts := client.PulsesOpts{Limit: optional.NewInt32(int32(limit))}
+		response, err := c.Pulses(t, &opts)
+		require.NoError(t, err)
+		require.Equal(t, response.Total, int64(pulsesNumber))
+		require.Equal(t, pulses[len(pulses)-1], response.Result[0].PulseNumber)
+		require.Len(t, response.Result, limit)
+	})
+	t.Run("limit max", func(t *testing.T) {
+		t.Log("C5174 Success limit validation, limit = 100")
+		limit := 100
+		opts := client.PulsesOpts{Limit: optional.NewInt32(int32(limit))}
+		response, err := c.Pulses(t, &opts)
+		require.NoError(t, err)
+		require.Equal(t, response.Total, int64(pulsesNumber))
+		require.Equal(t, pulses[len(pulses)-1], response.Result[0].PulseNumber)
+		require.Len(t, response.Result, limit)
+	})
+	t.Run("zero limit", func(t *testing.T) {
+		t.Log("C5172 Error limit validation, limit = 0")
+		limit, offset := 0, 10
+		opts := client.PulsesOpts{Offset: optional.NewInt32(int32(offset)), Limit: optional.NewInt32(int32(limit))}
+		_, err := c.Pulses(t, &opts)
+		require.Error(t, err)
+		require.Equal(t, "400 Bad Request", err.Error())
+	})
+	t.Run("too big limit", func(t *testing.T) {
+		t.Log("C5173 Error limit validation, limit = 101")
+		limit, offset := 101, 10
+		opts := client.PulsesOpts{Offset: optional.NewInt32(int32(offset)), Limit: optional.NewInt32(int32(limit))}
+		_, err := c.Pulses(t, &opts)
+		require.Error(t, err)
+		require.Equal(t, "400 Bad Request", err.Error())
+	})
+	t.Run("negative limit", func(t *testing.T) {
+		t.Log("C5210 Error limit validation, limit = -1")
+		limit, offset := 10, -1
```

Comment:
so there is offset is negative, we made mistake in test case name either in parameters.

Reply:
thanks, fixed